client-misk/src/test/kotlin/app/cash/backfila/client/misk/base/UnshardedHibernateBackfillTest.kt

```diff
@@ -0,0 +1,272 @@
+package app.cash.backfila.client.misk.base
+
+import app.cash.backfila.client.misk.ClientMiskService
+import app.cash.backfila.client.misk.ClientMiskTestingModule
+import app.cash.backfila.client.misk.DbMenu
+import app.cash.backfila.client.misk.EmbeddedBackfila
+import app.cash.backfila.client.misk.EmbeddedBackfillRun
+import app.cash.backfila.client.misk.MenuQuery
+import com.google.inject.Module
+import com.google.inject.TypeLiteral
+import misk.hibernate.Id
+import misk.hibernate.Query
+import misk.hibernate.Session
+import misk.hibernate.Transacter
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import okhttp3.internal.toImmutableList
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+import kotlin.test.assertEquals
+import kotlin.test.assertFalse
+import kotlin.test.assertNotNull
+import kotlin.test.assertNull
+import kotlin.test.assertTrue
+
+@MiskTest(startService = true)
+class UnshardedHibernateBackfillTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module: Module = ClientMiskTestingModule(false, listOf(TestBackfill::class))
+
+  @Inject @ClientMiskService lateinit var transacter: Transacter
+  @Inject lateinit var embeddedBackfila: EmbeddedBackfila
+
+  class TestBackfill @Inject constructor(
+    entityType: TypeLiteral<DbMenu>,
+    pkeyType: TypeLiteral<Id<DbMenu>>,
+    pkeySqlAdapter: PkeySqlAdapter,
+    @ClientMiskService transacter: Transacter,
+    queryFactory: Query.Factory
+  ) : UnshardedHibernateBackfill<DbMenu, Id<DbMenu>>(
+      entityType, pkeyType, pkeySqlAdapter, queryFactory, transacter) {
+    val idsRanDry = mutableListOf<Id<DbMenu>>()
+    val idsRanWet = mutableListOf<Id<DbMenu>>()
+    override fun backfillCriteria(config: BackfillConfig): Query<DbMenu> {
+      return queryFactory.newQuery(MenuQuery::class).name("chicken")
+    }
+    override fun runBatch(pkeys: List<Id<DbMenu>>, config: BackfillConfig) {
+      if (config.dryRun) {
+        idsRanDry.addAll(pkeys)
+      } else {
+        idsRanWet.addAll(pkeys)
+      }
+    }
+  }
+
+  @Test fun emptyTable() {
+    val run = embeddedBackfila.createDryRun(TestBackfill::class).configureForTest()
```

Comment:
Using a inline reified function would be pretty cool 👍

```
// definition
inline fun EmbeddedBackfila.createDryRun<reified T>() = createDryRun(T::class)

// usage
embeddedBackfila.createDryRun<TestBackfill>()
```

Reply:
Yeah I ended up doing this.